### PR TITLE
[WIP] add source-highlight

### LINKED ISSUE #5642: Package source-highlight
[source-highlight](https://www.gnu.org/software/src-highlite/) can be useful for documentation-building packages (see for example https://github.com/conda-forge/staged-recipes/pull/5487)

### PR DESCRIPTION
**Pull request by carlodri**
[GNU Source-highlight](https://www.gnu.org/software/src-highlite/), given a source file, produces a document with syntax highlighting. 

closes #5642

### CODE REVIEW

**Comment by carlodri** [comment]
@conda-forge/help-c-cpp can someone help me with this recipe? The compilation works fine, but when I test the script I get the following error:

```
source-highlight: error while loading shared libraries: libboost_regex.so.1.66.0: cannot open shared object file: No such file or directory
```

However, the file is actually present in `${PREFIX}/lib` so I can't really understand what's going on here...

I also tried to set the `LD_LIBRARY_PATH` variable to `$PREFIX/lib` but nothing changes.

File: recipes/source-highlight/build.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+export CXXFLAGS="${CXXFLAGS}"
+export LDFLAGS="${LDFLAGS}"
+
+ls -l ${PREFIX}/lib/libboost_regex*
```
Comment (dougalsutherland) [inline recipes/source-highlight/build.sh]: Remember to delete this. :)

File: recipes/source-highlight/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - toolchain
+    - boost-cpp
```
Comment (dougalsutherland) [inline recipes/source-highlight/meta.yaml]: Make it `boost-cpp 1.66.0`.

File: recipes/source-highlight/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    - libtool
+    - texinfo
+    - automake
+    - autoconf
+  run:
+    - boost-cpp
```
Comment (dougalsutherland) [inline recipes/source-highlight/meta.yaml]: Also `boost-cpp 1.66.0`.

File: recipes/source-highlight/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+{% set name = "source-highlight" %}
+{% set version = "3.1.8" %}
+{% set sha256 = "7cc44a905410d63b23f77877b3ceec98f07bedc086b2137880904ac8c1ad4877" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: http://git.savannah.gnu.org/cgit/src-highlite.git/snapshot/src-highlite-rel_{{ version }}.tar.gz
```
Comment (dougalsutherland) [inline recipes/source-highlight/meta.yaml]: This works, but probably better to use the official release tarball:
```
https://ftp.gnu.org/gnu/src-highlite/source-highlight-{{ version }}.tar.gz
```
which has sha256
`01336a7ea1d1ccc374201f7b81ffa94d0aecb33afc7d6903ebf9fbf33a55ada3`.

File: recipes/source-highlight/build.sh (hunk 1 above)
Comment (carlodri) [reply recipes/source-highlight/build.sh]: sure!

File: recipes/source-highlight/meta.yaml (hunk 4 above)
Comment (carlodri) [reply recipes/source-highlight/meta.yaml]: thanks, will do

**Comment by carlodri** [comment]
@dougalsutherland thanks for your review, silly me for not seeing that dependency issue...

**Comment by dougalsutherland** [comment]
Merged. Thanks @carlodri!

**Pull request merged**